src/nvim/tui/tui.c

```diff
@@ -1009,17 +1012,51 @@ static void tui_mouse_on(UI *ui)
 {
   TUIData *data = ui->data;
   if (!data->mouse_enabled) {
+#ifdef WIN32
+    bool is_conemu = terminfo_is_term_family(os_getenv("TERM"), "conemu");
+    uv_tty_vtermstate_t state;
+    uv_tty_get_vterm_state(&state);
+    bool is_legacy = state == UV_TTY_UNSUPPORTED;
+    bool use_vti = os_has_vti() && !is_conemu;
+    if (!is_legacy && !use_vti) {
+      flush_buf(ui);
+      uv_tty_set_vterm_state(UV_TTY_UNSUPPORTED);
+    }
+#endif
     unibi_out_ext(ui, data->unibi_ext.enable_mouse);
     data->mouse_enabled = true;
+#ifdef WIN32
+    if (!is_legacy && !use_vti) {
+      flush_buf(ui);
+      uv_tty_set_vterm_state(UV_TTY_SUPPORTED);
+    }
+#endif
   }
 }
 
 static void tui_mouse_off(UI *ui)
 {
   TUIData *data = ui->data;
   if (data->mouse_enabled) {
+#ifdef WIN32
+    bool is_conemu = terminfo_is_term_family(os_getenv("TERM"), "conemu");
+    uv_tty_vtermstate_t state;
+    uv_tty_get_vterm_state(&state);
+    bool is_legacy = state == UV_TTY_UNSUPPORTED;
+    bool use_vti = os_has_vti() && !is_conemu;
+    if (!is_legacy && !use_vti) {
+      flush_buf(ui);
+      uv_tty_set_vterm_state(UV_TTY_UNSUPPORTED);
+    }
+#endif
```

Comment:
- What is the difference between vterm and VTI? A reference/link would be helpful.
- Why special-case conemu?
- Why temporarily disable vterm before sending mouse on/off requests? Is this working around a problem with our termcodes?


This block (and same code in `tui_mouse_on`) should probably go in `os_win_console.c`, e.g. : 

```
void os_set_wintty(bool enable) {
  static bool is_conemu = ... // Only need to check once, won't to change at runtime.
  if (enable) {
    ...
  } else { 
    ... 
  }
}
```

Reply:
> - What is the difference between vterm and VTI? A reference/link would be helpful.
> - Why temporarily disable vterm before sending mouse on/off requests? Is this working around a problem with our termcodes?

Comment added. Is the description of the comment clear enough?

> - Why special-case conemu?

I installing and checking ConEmu, there was no need for a special case of ConEmu.

Reply:
yes, thanks!